### PR TITLE
feat: add precondition checks to file operations

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -251,6 +251,7 @@ export interface MoveCallback {
 
 export interface MoveOptions {
   userProject?: string;
+  preconditionOpts?: PreconditionOptions;
```
Comment (shaffeeullah) [inline src/file.ts:254]: Where is `PreconditionOptions` defined?

Comment (ddelgrosso1) [reply src/file.ts:254]: It was part of the earlier PR for gcs-resumable-upload: https://github.com/googleapis/nodejs-storage/blob/0c1fa3934a52a608366a8c6c798c43516dd03dbf/src/storage.ts#L61

**Comment by shaffeeullah** [comment]
This looks good to me! Thanks, Denis! Did you want to do them as separate releases? Merging to master will autorelease. If you'd like to do 1 release, you should merge to an intermediate branch first.

**Review by shaffeeullah — APPROVED** [review]

**Comment by ddelgrosso1** [comment]
@shaffeeullah I've been debating whether to do this as separate merges / releases. I suppose probably makes more sense to do it as a single release.

**Pull request merged**